src/ir/value.h

```diff
@@ -115,13 +139,22 @@ class Value {
   }
 
   // A downcast operation. Just delegates directly to std::get on variants.
-  template<class T>
-  const T &As() const { return std::get<T>(value_); }
+  template <class T>
+  const T& As() const {
+    return std::get<T>(value_);
+  }
 
   // A dynamic downcast operation. Just delegates directly to std::get_if on
   // variants.
-  template<class T>
-  const T *If() const { return std::get_if<T>(&value_); }
+  template <class T>
+  const T* If() const {
+    return std::get_if<T>(&value_);
+  }
+
+  // Return whether the two `Value`s are identical representations. If they
+  // are different, this does not necessarily mean that they do not
+  // *evaluate* to the same value.
+  bool operator==(const Value& other) const { return value_ == other.value_; }
```

Comment:
I presume this takes care of the variant kinds as well?

Reply:
Yes. From here: https://en.cppreference.com/w/cpp/utility/variant/operator_cmp

"""
If v.index() != w.index(), returns false;
"""

where `index()` returns the index of the alternative in the template parameter pack of alternatives.